internal/service/advancedclustertpf/resource_unit_test.go

```diff
@@ -0,0 +1,239 @@
+					resource.TestCheckResourceAttr(resourceName, "mongo_db_major_version", "8.0"),
+					resource.TestCheckResourceAttr(resourceName, "advanced_configuration.change_stream_options_pre_and_post_images_expire_after_seconds", "100"),
+				),
+			},
+			{
+				ResourceName:                         resourceName,
+				ImportStateIdFunc:                    acc.ImportStateIDFuncProjectIDClusterName(resourceName, "project_id", "name"),
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "name",
+			},
+		},
+	}
+	unit.MockTestCaseAndRun(t, vars, &unit.MockHTTPDataConfig{AllowMissingRequests: true, AllowReReadGet: true}, &testCase)
+}
+
+func shortenRetries() {
+	advancedclustertpf.RetryMinTimeout = 1 * time.Second
+	advancedclustertpf.RetryDelay = 1 * time.Second
+	advancedclustertpf.RetryPollInterval = 100 * time.Millisecond
+}
+
+func TestMockAdvancedCluster_configSharded(t *testing.T) {
+	var (
+		projectID   = "111111111111111111111111"
+		clusterName = "sharded-multi-replication"
+		vars        = map[string]string{
+			"groupId":     projectID,
+			"clusterName": clusterName,
+		}
+	)
+	shortenRetries()
+	testCase := resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		Steps: []resource.TestStep{
+			{
+				Config: configSharded(projectID, clusterName, false),
+				Check:  resource.TestCheckResourceAttr(resourceName, "project_id", projectID),
+			},
+			{
+				Config: configSharded(projectID, clusterName, true),
+				Check:  resource.TestCheckResourceAttr(resourceName, "name", clusterName),
+			},
+			{
+				ResourceName:                         resourceName,
+				ImportStateIdFunc:                    acc.ImportStateIDFuncProjectIDClusterName(resourceName, "project_id", "name"),
+				ImportState:                          true,
+				ImportStateVerify:                    true,
+				ImportStateVerifyIdentifierAttribute: "name",
+			},
+		},
+	}
+	unit.MockTestCaseAndRun(t, vars, &unit.MockHTTPDataConfig{AllowMissingRequests: true, AllowReReadGet: true}, &testCase)
+}
+
+func configSharded(projectID, clusterName string, withUpdate bool) string {
+	var autoScaling, analyticsSpecs string
+	if withUpdate {
+		autoScaling = `
+			auto_scaling = {
+				disk_gb_enabled = true
+			}`
+		analyticsSpecs = `
+			analytics_specs = {
+				instance_size   = "M30"
+				node_count      = 1
+				ebs_volume_type = "PROVISIONED"
+				disk_iops       = 2000
+			}`
+	}
+	// SDK v2 Implementation receives many warnings, one of them: `.replication_specs[1].region_configs[0].analytics_specs[0].disk_iops: was cty.NumberIntVal(2000), but now cty.NumberIntVal(1000)`
+	// Therefore, in TPF we are forced to set the value that will be returned by the API (1000)
+	// The rule is: For any replication spec, the `(analytics|electable|read_only)_spec.disk_iops` must be the same across all region_configs
+	// The API raises no errors, but the response reflects this rule
```

Comment:
should we confirm with Atlas team?

Reply:
I prefer syncing once I have a list of all the inconsistencies

Reply:
sounds good